Mathlib/CategoryTheory/Products/Bifunctor.lean

```diff
@@ -46,4 +45,58 @@ theorem diagonal' (F : C × D ⥤ E) (X X' : C) (f : X ⟶ X') (Y Y' : D) (g : Y
     F.map (f ×ₘ 𝟙 Y) ≫ F.map (𝟙 X' ×ₘ g) = F.map (f ×ₘ g) := by
   rw [← Functor.map_comp, prod_comp, Category.id_comp, Category.comp_id]
 
+/-- Opposite of a bifunctor -/
+@[simps!]
+def biop (F : C × D ⥤ E) : Cᵒᵖ × Dᵒᵖ ⥤ Eᵒᵖ :=
+  (prodOpEquiv C).inverse ⋙ F.op
+
+/-- Flipping of arguments of a bifunctor -/
+@[simps!]
+def flip (F : C × D ⥤ E) : D × C ⥤ E :=
+  Prod.swap D C ⋙ F
+
 end CategoryTheory.Bifunctor
+
+namespace CategoryTheory.Functor
+
+/-- Opposite of a bifunctor. -/
+@[simps!]
+def biop (F : Cᵒᵖ ⥤ D ⥤ E) : C ⥤ Dᵒᵖ ⥤ Eᵒᵖ := F.rightOp ⋙ Functor.opHom _ _
+
+end CategoryTheory.Functor
+
+namespace CategoryTheory
+
+variable {C₁ C₂ : C} {D₁ D₂ : D} {E₁ E₂ : E}
+
+/-- Action of two-variable functors on objects. -/
+abbrev Functor.obj₂ (H : C ⥤ D ⥤ E) (A : C) (B : D) : E := (H.obj A).obj B
+
+/-- Action of three-variable functors on objects. -/
+abbrev Functor.obj₃ (H : C ⥤ D ⥤ E ⥤ F) (A : C) (B : D) (C : E) : F :=
+  ((H.obj A).obj B).obj C
+
+/-- Apply a natural transformation between bifunctors to two objects. -/
+abbrev NatTrans.app₂ {F G : C ⥤ D ⥤ E} (α : NatTrans F G) (X : C) (Y : D) :
```

Comment:
```suggestion
abbrev NatTrans.app₂ {F G : C ⥤ D ⥤ E} (α : F ⟶ G) (X : C) (Y : D) :
```
I think we prefer to avoid explicit mentions of `NatTrans` as soon as the category structure on functors is available (i.e almost always), even if it’s def-eq (usually, this helps lean recognizing α as a morphism in some category).
There’s a similar comment below.